src/overlays/actors/ovl_Bg_Gnd_Darkmeiro/z_bg_gnd_darkmeiro.c

```diff
@@ -35,25 +35,225 @@ const ActorInit Bg_Gnd_Darkmeiro_InitVars = {
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F); //set switch N.
+    } else { //If both are under 0x40,
+		Flags_UnsetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F); //clear switch N.
+	} //Switch N is cleared 0x40 frames early to give the transparent platforms time to flicker and vanish in sync with the other switches. This flickering effect is bugged, so it doesn't happen.
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_PlatformTimer.s")
+
+
+void BgGndDarkmeiro_StaticPlatform(BgGndDarkmeiro *this, GlobalContext *globalCtx) {
+	//Static platforms do nothing, just like this function.
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_StaticPlatform.s")
+
+void BgGndDarkmeiro_SwitchedPlatform(BgGndDarkmeiro *this, GlobalContext *globalCtx) {
+
+    if (this->timer1 > 0) { //If the vanishing timer isn't 0, decrement it.
+		this->timer1 -= 1;
+    }
+	
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F) != 0) {
+        this->timer1 = 0x40; //If the switch is on, reset the vanishing timer.
+    }
+	
+    BgGndDarkmeiro_TogglePlatform(this, globalCtx); //Set whether the platform is solid.
+}
+
+
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_SwitchedPlatform.s")
+
+void BgGndDarkmeiro_Update(Actor *thisx, GlobalContext *globalCtx) {
+	BgGndDarkmeiro* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_Destroy.s")
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_Update.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/func_80878F28.s")
+void BgGndDarkmeiro_Draw0(BgGndDarkmeiro *this, GlobalContext *globalCtx) {
+    Gfx_DrawDListXlu(globalCtx, &D_060088B0); //Draws...something for mode 0.
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/func_80878F34.s")
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_Draw0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/func_80879128.s")
+void BgGndDarkmeiro_DrawSwitchedPlatform(BgGndDarkmeiro *this, GlobalContext *globalCtx) {
+	s16 newvar;
+	s32 newvar2;
+    GraphicsContext *gfxCtx;
+	Gfx *dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/func_80879134.s")
+	newvar = this->timer1;
+    if (newvar != 0) { // Platform only gets drawn if vanishing timer isn't zero.
+        if (newvar >= 0x41) {
+			this->timer2 = (this->timer2 < 0x78) ? this->timer2 + 8 : 0x7F;
+        } else if (newvar >= 0x11) {
+			newvar2 = this->timer1 ;
+			this->timer2 =  (Math_Coss( (s16) newvar2 * 4096 ) * 64.0f) + 127.0f;
+			if (this->timer2 >= 0x80) {
+				this->timer2 = 0x7F;
```

Comment:
This doesn't matter for matching but I believe `>` makes more sense than `>=` here according to the values.
```suggestion
        if (newvar > 0x40) {
			this->timer2 = (this->timer2 < 0x78) ? this->timer2 + 8 : 0x7F;
        } else if (newvar > 0x10) {
			newvar2 = this->timer1 ;
			this->timer2 =  (Math_Coss( (s16) newvar2 * 4096 ) * 64.0f) + 127.0f;
			if (this->timer2 > 0x7F) {
				this->timer2 = 0x7F;
```